doc/sphinxdoc/generate_reference.py

```diff
@@ -160,6 +187,15 @@ def doc2rst(algo_doc, sphinxsearch=False):
         lines = [ algo_doc['name'], '=' * len(algo_doc['name']), '']
         lines += [algo_doc['mode'] + ' | ' + algo_doc['category'] + ' category', '']
 
+    links = source_links(algo_doc)
+    if links:
```

Comment:
I suggest moving this section to put it between the "Description" and "See also" sections because:
- source code makes more sense after reading basic algorithm description
- not everyone will want to inspect C++ source code